src/main/java/dao/PieceDao.java

```diff
@@ -0,0 +1,129 @@
+package dao;
+
+import domain.position.Position;
+import domain.unit.Team;
+import domain.unit.UnitType;
+import entity.Piece;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PieceDao {
+    private static final String SERVER = "localhost:13306"; // MySQL 서버 주소
+    private static final String DATABASE = "janggi"; // MySQL DATABASE 이름
+    private static final String OPTION = "?useSSL=false&allowPublicKeyRetrieval=true&serverTimezone=UTC";
+    private static final String USERNAME = "root"; //  MySQL 서버 아이디
+    private static final String PASSWORD = "root"; // MySQL 서버 비밀번호
+
+    public Connection getConnection() {
+        try {
+            return DriverManager.getConnection("jdbc:mysql://" + SERVER + "/" + DATABASE + OPTION, USERNAME, PASSWORD);
+        } catch (final SQLException e) {
+            System.err.println("DB 연결 오류:" + e.getMessage());
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public void save(final Piece piece, final String roomId) {
+        final var query = """
+                INSERT INTO piece(position_x, position_y, piece_type, team, room_id)
+                VALUES (?, ?, ?, ?, ?)
+                """;
+        try (final var connection = getConnection();
+             final var ppst = connection.prepareStatement(query)) {
+            ppst.setInt(1, piece.positionX());
+            ppst.setInt(2, piece.positionY());
+            ppst.setString(3, piece.unitType().toString());
+            ppst.setString(4, piece.team().toString());
+            ppst.setString(5, roomId);
+            ppst.executeUpdate();
```

Comment:
executeUpdate()의 반환값을 체크해보면 어떨까요? 😃  (반환 결과가 어떤 값인지 먼저 알아보신 뒤)
이 값을 통해서 적절히 예외를 던져주면 업데이트가 실패했음을 감지할 수 있을 것 같아요. 🤔

```suggestion
            int rowsUpdated = ppst.executeUpdate();
            if (rowsUpdated == 0) {
                throw new SQLException("No rows updated");
            }
}
```

Reply:
> executeUpdate()에 대한 반환값을 확인해서 0이면 예외 발생시키는 건에 대해서는 반영을 하지 않았습니다.
만약 조건문에 맞는 레코드가 없어 변경할게 없는 상황을 생각해보면, 예외가 아닌 당연한 상황이라고 생각했습니다.

오.. 그렇군요. 업데이트가 꼭 필요하지 않은 경우에도 업데이트 수행을 할 수 있는 상황이란걸 인지하지 못했어요. 😅 

좋은 지적이신 것 같습니다. 👍👍 
잘 파악해주셨네요. 🤗